### DISCUSSION TITLE
Webhooks

### DISCUSSION THREAD

**Discussion by krestaino** [navidrome/navidrome#2784] [Ideas]
Specifically, a way for firing webhooks when something happens on the server, such as adding new music, deleting music, playing songs, etc., and allowing us to send data regarding the event to an arbitrary endpoint.

I'm curious if there is anyone else who would find value in this. I recently posted a [project](https://github.com/navidrome/navidrome/discussions/2773) I'm working on and would like to expand it by adding some automation. Ideally, if Navidrome could fire a webhook with album IDs that have just been added, which I can listen to, I could immediately set the popularity ratings for the new music.

Outside of my specific use case, I could see adding webhooks being useful in other ways. For example, sending notifications or emails to users when new music is added, or setting your Discord status to the song you're currently listening to. I'm sure there are many cool things people could do or integrate if there were webhooks in Navidrome.

I found this [issue](https://github.com/navidrome/navidrome/issues/478), though it's been closed. I really like the idea, so I wanted to post here, seeking out any other ideas that could help justify such a feature.

**Comment by Proselyte-M** [comment]
i think i need this 功能
